File: editor/en/[118]Pascal's Triangle.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function (numRows) {
    let res = new Array(numRows);
    for (let i = 0; i < numRows; i++) {
        res[i] = new Array();
        for (let j = 0; j <= i; j++) {
            if (j === 0 || j === i) {
                res[i][j] = 1;
            } else {
                res[i][j] = res[i - 1][j - 1] + res[i - 1][j];
            }
        }
    }
    return res;
};
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[118]Pascal's Triangle.java
118
        Pascal's Triangle
        2022-12-22 00:47:10

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> result = new ArrayList<>();

        for (int i = 0; i < numRows; i++) {
            List<Integer> list = new ArrayList<>();

            for (int j = 0; j <= i; j++) {
                if (j == 0 || j == i) {
                    list.add(1);
                } else {
                    list.add(result.get(i - 1).get(j - 1) + result.get(i - 1).get(j));
                }
            }

            result.add(list);
        }

        return result;
    }
}
//leetcode submit region end(Prohibit modification and deletion)
